src/components/picker/__tests__/index.spec.tsx

```diff
@@ -100,122 +96,191 @@ describe('Picker', () => {
       });
     });
 
-    it('Test onDismiss', () => {
-      const driver = getDriver({
-        pickerModalProps: {
-          onDismiss
-        }
-      });
-
-      expect(driver.isOpen()).toBeFalsy();
+    it('Test onPress', () => {
+      const driver = getDriver({onPress});
       driver.open();
       expect(driver.isOpen()).toBeTruthy();
       driver.cancel();
       expect(driver.isOpen()).toBeFalsy();
-      expect(onDismiss).toHaveBeenCalledTimes(2); // TODO: this should be 1
+      expect(onPress).toHaveBeenCalled();
+    });
+
+    describe('Test pickerModalProps', () => {
+      it('Test onDismiss', () => {
+        const driver = getDriver({
+          pickerModalProps: {
+            onDismiss
+          }
+        });
+        driver.open();
+        expect(driver.isOpen()).toBeTruthy();
+        driver.cancel();
+        expect(driver.isOpen()).toBeFalsy();
+        expect(onDismiss).toHaveBeenCalledTimes(2); // TODO: this should be 1
+      });
+
+      // TODO: this test is not passing yet
```

Comment:
When should this start working?

Reply:
The issue here is that the `onShow` function is triggered only when the modal is fully open.
Iv'e tried to use `act` and `waitFor` didn't help.

I'll change the `TODO` comment to fix this test in the future, WDYT ?

Reply:
Doesn't matter IMO